R/hiper_glm.R

```diff
@@ -1,31 +1,44 @@
 #' @export hiper_glm
-#' @export coef.hglm
-#' @export vcov.hglm
-#' @export print.hglm
+hiper_glm <- function(design, outcome, model='linear', option = list()){
 
-hiper_glm <- function(design, outcome, model='linear'){
+  ## Check validity
   supported_models <- c('linear')
-  if(!(model %in%  supported_models)){stop(sprintf("The model %s is not supported.", model))
-}
-  
-  warning("The function is yet to be implemented")
-  ## TODO: 
-  hglm_out <- list()
+  if(!(model %in%  supported_models)){stop(sprintf("The model %s is not supported.",
+                                                   model))
+  }
+
+
+  if (is.null(option$mle_solver)) {
```

Comment:
You'll probably want to do conditionals for solvers within each model, as opposed to choosing the model within each solver, since the solvers may differ by model (e.g. no pinv for logit model)